### ISSUE TITLE
failure to import scraper code when running tests in a certain way

### ISSUE THREAD

**Issue by bepetersn** [sc3/cookcountyjail#302]
For documentation's sake, I'm writing this out, @nwinklareth. Confirmed: from a clean repo, clean virtualenv, on Debian Linux 7, running `py.test`, which ought to execute our testsuite, sometimes fails with strange import errors.

This also happens on our [travis build](https://travis-ci.org/sc3/cookcountyjail/builds/21349471). For ease of reference, I have included a link to that failed build, where you can see 8 import errors.

**Comment by bepetersn** [comment]
As @nwinklareth pointed out, this doesn't happen if you make the very simple change of running `python -m py.test` instead of just `py.test`. We have made this change in the Travis build, and I made a recommendation on how to test in the README. Still, it'd be nice to clear up why this is happening.

**Comment by bepetersn** [comment]
This is related to [PEP 366](http://legacy.python.org/dev/peps/pep-0366/), I believe.

**Comment by bepetersn** [comment]
PEP 366 explains why using the `-m` approach works, but I figured out that it's actually just a python path issue. pytest doesn't include the current working directory, but you can have your virtualenv set this up for you with the command:

`add2virtualenv .`

This assumes you are in the project directory, in your virtualenv, and and have the virtualenvwrapper tools installed, of which this command is one.

**Comment by bepetersn** [comment]
See [this](http://stackoverflow.com/questions/20971619/ensuring-py-test-includes-the-application-directory-in-sys-path) stackoverflow question, which gives several options of how to fix this. Basically, it's a path issue. Your application/project needs to be on the path for pytest to know what to do, and it's intentional. 

On my system, I ran `add2virtualenv .` in my project directory, while in my venv, and everything just worked. I added this line to the scraper.sh script as well. 

But there are other ways to get it to work, as that question shows.
